Cql/CoreTests/PrimitiveTests.cs

```diff
@@ -3449,6 +3449,30 @@ public void Sort_Lists_Dates_Containing_Null()
             Assert.AreEqual(null, descending[2]);
         }
 
+        [TestMethod]
+        public void Collapse_Properly_Included()
+        {
+            var rtx = GetNewContext();
+
+            var expected = new[] {
+                new CqlInterval<CqlDate>(new CqlDate(2023, 1, 20), new CqlDate(2023, 1, 28),true, true),
+                new CqlInterval<CqlDate>(new CqlDate(2023, 2, 18), new CqlDate(2023, 2, 28),true, true),
+            };
+
+            var intervals = new[]
+            {
+                new CqlInterval<CqlDate>(new CqlDate(2023, 1, 20), new CqlDate(2023, 1, 28), true, true),
+                new CqlInterval<CqlDate>(new CqlDate(2023, 1, 22), new CqlDate(2023, 1, 25), true, true),
+                new CqlInterval<CqlDate>(new CqlDate(2023, 2, 20), new CqlDate(2023, 2, 25), true, true),
+                new CqlInterval<CqlDate>(new CqlDate(2023, 2, 18), new CqlDate(2023, 2, 28), true, true)
+            };
+
+            var collapsed = rtx.Operators.Collapse(intervals, null).ToArray();
+            var result = rtx.Operators.Comparer.Compare(expected!, collapsed!, null);
+            if (result != 0)
+                throw new AssertFailedException($"Expected {expected}; actual {collapsed}");

```

Comment:
What are the advantages of throwing over just `Assert.AreEqual(0, result)` ?

Reply:
No advantages that I know of. I used Collapse_Containing_Null as a template for building this test to follow convention and that uses AssertFailedException as well.

Reply:
This looks weird to me, but if it was a standing practice, I guess we should not bother about this case eithere.